### PR TITLE
add common test and build action

### PR DESCRIPTION
**Pull request by fairclothjm**

### CODE REVIEW

**Review by swenson — APPROVED** [review]

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Tests
+
+on: [push, workflow_dispatch]
```
Comment (swenson) [inline .github/workflows/tests.yaml:3]: I think this isn't quite right -- it's needs to be `workflow_call` I think?

**Review by swenson — DISMISSED** [review]
Oops, dismissing my previous review for now.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (fairclothjm) [reply .github/workflows/tests.yaml]: Oops! Thanks! done in https://github.com/hashicorp/vault-workflows-common/pull/5/commits/137b6bfb76e0506b045aa9b81aea1a3cc6483b44

**Review by maxcoulombe — APPROVED** [review]
Thanks!

**Review by swenson — COMMENTED** [review]
Sorry about that, I'll remember not to use request changes in the future!

LGTM

**Comment by fairclothjm** [comment]
@swenson No problem! Thanks!

**Pull request merged**